scripts/e2e-compatibility-matrix.sh

```diff
@@ -0,0 +1,200 @@
+# Copyright 2025 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/usr/bin/env bash
+# KubeStellar/KubeFlex Compatibility E2E Matrix Test Script
+# Migrated from GitHub Actions to run in Prow/OCI. Fetches matrix dynamically.
+set -euo pipefail
+
+retry() {
+  local -r -i max_attempts="$1"; shift
+  local -r cmd=("$@")
+  local -i attempt_num=1
+  local delay=5
+  until "${cmd[@]}"; do
+    if (( attempt_num == max_attempts )); then
+      echo "[ERROR] Command '${cmd[*]}' failed after $attempt_num attempts."
+      return 1
+    else
+      echo "[WARN] Command '${cmd[*]}' failed. Retrying in $delay seconds... ($attempt_num/$max_attempts)"
+      sleep $delay
+      ((attempt_num++))
+      delay=$((delay * 2))
+    fi
+  done
+}
+
+# Get min required kflex version from script
+MIN_KFLEX_VERSION=$(python3 scripts/extract_min_kflex_version.py || true)
+if [[ -z "$MIN_KFLEX_VERSION" ]]; then
+  echo "[ERROR] Failed to determine minimum kflex version."
+  exit 1
+fi
+echo "[INFO] Minimum kflex version required: $MIN_KFLEX_VERSION"
+
+# Prepare GitHub API authorization header if GITHUB_TOKEN is set
+AUTH_HEADER=()
+if [[ -n "${GITHUB_TOKEN:-}" ]]; then
+  AUTH_HEADER=(-H "Authorization: token $GITHUB_TOKEN")
+fi
+
+# Fetch all kflex releases >= min version
+RELEASES_JSON=$(retry 5 curl -s "${AUTH_HEADER[@]}" https://api.github.com/repos/kubestellar/kubeflex/releases)
+KUBEFLEX_VERSIONS=()
+mapfile -t ALL_VERSIONS < <(echo "$RELEASES_JSON" | jq -r '.[].tag_name' | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V)
+for V in "${ALL_VERSIONS[@]}"; do
+  VNUM=${V#v}
+  if [[ "$(printf "%s\n%s" "$MIN_KFLEX_VERSION" "$VNUM" | sort -V | head -1)" == "$MIN_KFLEX_VERSION" ]]; then
```

Comment:
Wouldn't it be simpler to write `if ! [[ "$VNUM" < "$MIN_KFLEX_VERSION" ]]; then` ?